tests/unit/app_integrations/test_apps/test_slack.py

```diff
@@ -0,0 +1,399 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+# pylint: disable=abstract-class-instantiated,protected-access,no-self-use,abstract-method,anomalous-backslash-in-string
+from mock import Mock, patch
+from nose.tools import assert_equal, assert_false, assert_items_equal, raises
+
+from app_integrations.apps.slack import SlackApp, SlackAccessApp, SlackIntegrationsApp
+from app_integrations.config import AppConfig
+
+
+from tests.unit.app_integrations.test_helpers import (
+    get_valid_config_dict,
+    MockSSMClient
+    )
+
+@patch.object(AppConfig, 'SSM_CLIENT', MockSSMClient())
+@patch.object(SlackApp, '_endpoint', Mock(return_value='endpoint'))
+@patch.object(SlackApp, '_type', Mock(return_value='type'))
+class TestSlackApp(object):
+    """Test class for the SlackAccessApp"""
+
+    def __init__(self):
+        self._app = None
+
+    @patch.object(SlackApp, '__abstractmethods__', frozenset())
+    def setup(self):
+        self._app = SlackApp(AppConfig(get_valid_config_dict('slack')))
+
+    def test_required_auth_info(self):
+        """SlackApp - Required Auth Info"""
+        assert_items_equal(self._app.required_auth_info().keys(), {'auth_token'})
+
+    def test_date_formatter(self):
+        """SlackApp - Date Formatter"""
+        assert_equal("%s", self._app.date_formatter())
+
+    @patch('requests.post')
+    @patch('logging.Logger.exception')
+    def test_error_code_return(self, log_mock, requests_mock):
+        """SlackApp - Gather Logs - Bad Response"""
+        requests_mock.return_value = Mock(
+            status_code=404
+            )
+        assert_false(self._app._gather_logs())
+        log_mock.assert_called_with('Received bad response from slack')
+
+    @patch('requests.post')
+    @patch('logging.Logger.exception')
+    def test_error_response_from_slack(self, log_mock, requests_mock):
+        """SlackApp - Gather Logs - Error Response"""
+        requests_mock.return_value = Mock(
+            status_code=200,
+            json=Mock(return_value={'ok':False, 'error':'paid_only'})
+            )
+        assert_false(self._app._gather_logs())
+        log_mock.assert_called_with('Received error or warning from slack')
+
+
+@patch.object(AppConfig, 'SSM_CLIENT', MockSSMClient())
+class TestSlackAccessApp(object):
+    """Test class for the SlackAccessApp"""
+
+    def __init__(self):
+        self._app = None
+
+    @patch.object(SlackAccessApp, '__abstractmethods__', frozenset())
+    def setup(self):
+        self._app = SlackAccessApp(AppConfig(get_valid_config_dict('slack')))
+
+    def test_sleep_seconds(self):
+        """SlackAccessApp - Sleep Seconds"""
+        assert_equal(3, self._app._sleep_seconds())
+
+    @staticmethod
+    def _get_sample_access_logs():
+        """Sample logs collected from the slack api documentation"""
+        return {
```

Comment:
a bit of a nit, but to be consistent with the codebase: can you ensure all string literals use single quotes (`'`) not double quotes? and omit the leading `u` prefix. there are a few other instances within this file as well.